common/src/app/common/types/tokens_lib.cljc

```diff
@@ -60,14 +54,38 @@
 (defn get-path
   "Get the path of object by specified separator (E.g. with '.' separator, the
   'group.subgroup.name' -> ['group' 'subgroup'])"
-  [item separator]
-  (assert (valid-groupable-item? item) "expected groupable item")
-  (->> (split-path (:name item) separator)
+  [name separator]
+  (->> (split-path name separator)
        (not-empty)))
 
+;; === Common
+
+(defprotocol INamedItem
+  "Protocol for items that have a name, a description and a modified date."
+  (get-name [_] "Get the name of the item.")
+  (get-description [_] "Get the description of the item.")
+  (get-modified-at [_] "Get the description of the item.")
+  (rename [_ new-name] "Set the name of the item.")
```

Comment:
I think this should be consistent with the others, if we use the `set-...` idiom, i think we should use it also for name.

Reply:
I thought about it. But changing the name has more implications than just setting the value of an attribute (i.e. it may change the position in the tree or the priority of a set). This is the reason I chose to give a name more suggestive.